tests/integration/test_record_writer.py

```diff
@@ -299,3 +303,120 @@ def test_decoding_line_numbers(tmp_path):
             ("bar", "bar b.py", 16),
             ("foo", "foo a.py", 5),
         ]
+
+
+@pytest.mark.skipif(
+    sys.version_info < (3, 13), reason="Object tracking requires Python 3.13+"
+)
```

Comment:
As above, there's no reason to skip this on <3.13, we should just allow the test record writer to write these records for all versions.
```suggestion
```